plugins/SSH.koplugin/main.lua

```diff
@@ -0,0 +1,118 @@
+local Device =  require("device")
+local InfoMessage = require("ui/widget/infomessage")  -- luacheck:ignore
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("util")
+local _ = require("gettext")
+
+local SSH = WidgetContainer:new{
+    name = 'SSH',
+    is_doc_only = false,
+}
+
+function SSH:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function SSH:addToMainMenu(menu_items)
+    local SSH_port = G_reader_settings:readSetting("SSH_port") or "2222"
+    local allow_blank_password = false
+    local start = function(menu)
```

Comment:
Most of this stuff should probably be separated out above instead of being inlined in addToMainMenu. The scope will still be restricted to SSH.koplugin